ratis-shell/src/main/java/org/apache/ratis/shell/cli/sh/snapshot/TakeSnapshotCommand.java

```diff
@@ -50,16 +52,22 @@ public String getCommandName() {
   public int run(CommandLine cl) throws IOException {
     super.run(cl);
     long timeout;
+    final RaftPeerId peerId;
     if (cl.hasOption(TAKE_SNAPSHOT_TIMEOUT_OPTION_NAME)) {
       timeout = Long.parseLong(cl.getOptionValue(TAKE_SNAPSHOT_TIMEOUT_OPTION_NAME));
     } else {
       timeout = 3000;
     }
     try(final RaftClient raftClient = RaftUtils.createClient(getRaftGroup())) {
-      RaftClientReply reply = raftClient.getSnapshotManagementApi().create(timeout);
-      processReply(reply, () -> String.format("Failed to take snapshot of peerId %s", raftClient.getLeaderId()));
-      printf(String.format("Successful take snapshot, the latest snapshot index is %d",
-          reply.getLogIndex()));
+      if (cl.hasOption(PEER_ID_OPTION_NAME)) {
+        peerId = RaftPeerId.getRaftPeerId(cl.getOptionValue(PEER_ID_OPTION_NAME));
+      } else {
+        peerId = raftClient.getLeaderId();
```

Comment:
Just set peerId = null.  The leader could change so that we should getLeaderId right before sending the rpc.